### PR TITLE
openmpi: added singularity variant

### PR DESCRIPTION
**Pull request by ajkotobi**

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]
@hppritcha can you review?

**Comment by hppritcha** [comment]
I tried to start a review but that didn't work.
there's a spelling mistake and just a heads up that the singularity configure option will go away in Open MPI 5

**Comment by ajkotobi** [comment]
@hppritcha , I added version condition for singularity with OpenMPI and pushed the changes to this PR. Where is the misspelling?

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 1]
```diff
@@ -219,6 +219,8 @@ class Openmpi(AutotoolsPackage):
     variant('cxx', default=False, description='Enable C++ MPI bindings')
     variant('cxx_exceptions', default=False, description='Enable C++ Exception support')
     variant('gpfs', default=True, description='Enable GPFS support (if present)')
+    variant('singularity', default=False,
+            description="Build support for the Singularity containe")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openmpi/package.py]:
```suggestion
            description="Build support for the Singularity container")
```

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 2]
```diff
@@ -265,6 +267,8 @@ class Openmpi(AutotoolsPackage):
     depends_on('sqlite', when='+sqlite3@:1.11')
     depends_on('zlib', when='@3.0.0:')
     depends_on('valgrind~mpi', when='+memchecker')
+    # Singularity release 3 works better
+    depends_on('singularity@3.0.0:', when='+singularity')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openmpi/package.py:271]: You should also add the following conflict below:
```python
conflicts('+singularity', when='@5:', msg='singularity support has been dropped in OpenMPI 5')
```

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 3]
```diff
@@ -528,6 +532,11 @@ def configure_args(self):
                 '--with-valgrind={0}'.format(spec['valgrind'].prefix),
             ])
 
+        # Singularity container support
+        if spec.satisfies('@:4.9'):
+	    if '+singularity' in spec:
+	       config_args.append(
+                 '--with-singularity={0}'.format(spec['singularity'].prefix))
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/openmpi/package.py]: Wouldn't a single condition:
```suggestion
        # Singularity container support
        if spec.satisfies('+singularity @:4.9'):
            singularity_opt = '--with-singularity={0}'.format(spec['singularity'].prefix)
            config_args.append(singularity_opt)
```
work as well?

Comment (ajkotobi) [reply var/spack/repos/builtin/packages/openmpi/package.py]: That worked, wasn't aware of this way in `spec`, thanks

**Pull request merged**